packages/server/mddleware/authMiddleware.ts

```diff
@@ -0,0 +1,46 @@
+import type { Request, Response, NextFunction } from 'express';
+import axios from 'axios';
+import type { User } from '../types/express-augment.js';
+
+const AUTH_CHECK_URL = 'https://ya-praktikum.tech/api/v2/auth/user';
+
+export const authMiddleware = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> => {
+  try {
+    const cookies = req.cookies;
+    const authHeader = req.headers.authorization;
+
+    const headers: Record<string, string> = {};
+
+    if (cookies && Object.keys(cookies).length > 0) {
+      const cookieString = Object.entries(cookies)
+        .map(([key, value]) => `${key}=${value}`)
```

Comment:
 при формировании cookieString нет экранирования значений. Если cookies содержат специальные символы, это может привести к ошибкам.
```suggestion
        .map(([key, value]) => `${key}=${encodeURIComponent(value)}`)
```